odoo/tools/misc.py

```diff
@@ -363,6 +363,27 @@ def add_sheet(self, name, cell_overwrite_ok=False):
     xlwt = None
 
 
+try:
+    import xlsxwriter
+
+    # add some sanitizations to respect the excel sheet name restrictions
+    # as the sheet name is often translatable, can not control the input
+    class PatchedXlsxWorkbook(xlsxwriter.Workbook):
+
+        def add_worksheet(self, name=None):
```

Comment:
the [original method](https://github.com/jmcnamara/XlsxWriter/blob/b512eabddb8f4564a83d469b0522d2f889787baa/xlsxwriter/workbook.py#L165) has a default parameter `worksheet_class=None`. It is a good idea to keep it (and avoid needing to do something like 007607030a2c810dc7b134e926a8203b94babf84 afterwards)